atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -326,26 +326,23 @@ public void registerClosingCallback(Runnable closingCallback) {
      */
     @Override
     public void close() {
-        if (isClosed.compareAndSet(false, true)) {
-            super.close();
-            cleaner.close();
-            keyValueService.close();
-            shutdownExecutor(deleteExecutor);
-            shutdownExecutor(getRangesExecutor);
-            closeLockServiceIfPossible();
-
-            List<Throwable> suppressedExceptions = new ArrayList<>();
+        if (!isClosed.compareAndSet(false, true)) {
+            return;
+        }
+
+        try (ShutdownRunner shutdownRunner = new ShutdownRunner()) {
+            shutdownRunner.shutdownSafely(super::close);
```

Comment:
Diff here is much better in IntelliJ. The order should be exactly the same, just wrapped with shutdownSafely calls